app/static/src/app/components/options/TagInput.vue

```diff
@@ -36,21 +36,24 @@ export default defineComponent({
 
         const paramValues = computed(() => store.state.run.parameterValues);
 
-        watch(paramValues, () => {
-            if (props.tags) {
-                emit("update", [...props.tags]);
-            }
-        }, { deep: true });
-
-        const computedTags = computed(() => {
+        const cleanTags = computed(() => {
             if (props.tags?.length === 0 || !props.tags) {
                 return [];
             }
-            const cleanTags = props.tags.filter((tag) => {
+            return props.tags.filter((tag) => {
                 if (typeof tag === "number") return true;
                 return !props.numericOnly && paramValues.value[tag] !== undefined;
             });
-            return cleanTags.map((tag) => {
+        });
+
+        onMounted(() => {
+            if (props.tags && cleanTags.value.length !== props.tags.length) {
+                emit("update", cleanTags.value);
+            }
+        });
```

Comment:
Does this not want to be in a watch on `paramValues` rather than just in onMounted? Or does it not need to be because in practice we don't actually change the available parameter names in the tab where `TagInput` appears, and changed to parameter values are now handled in the new `computedTags`?

Reply:
so the reason this was changed is because of an issue with the fit tab! the fit tab changes the paramValues so you are constantly in a state of updateing advanced settings so youll always have the user message of advanced settings have been changed, please run the model again :laughing: I switched to on mount as the only thing this was doing is making sure if the user removed a variable from their code and re-compiled, the variable would be deleted from the tag array in our state

Reply:
and the values are automatically updated in the computed tags anyway!

Reply:
Ok, thought that was probably it.